packages/storybook/src/generators/configuration/util-functions.ts

```diff
@@ -533,3 +534,22 @@ export function rootFileIsTs(
     return tsConfiguration;
   }
 }
+
+export function getE2EProjectName(
+  tree: Tree,
+  project: string
+): string | undefined {
+  const projects = getProjects(tree);
+  for (const [projectName, projectConfig] of projects.entries()) {
```

Comment:
This should use `forEachExecutorOptions`. Otherwise, this doesn't handle when projects don't have any `targets`.

```suggestion
  let e2eProject: string;
  forEachExecutorOptions(tree, '@nrwl/cypress:cypress', (options) => {
    if (e2eProject) {
      return;
    }
    
    if (options.devServerTarget === `${project}:serve` || ...) {
      e2eProject = projectName;
    }
  });
  return e2eProject;
```

Reply:
Sure, I'll change it! I just copied the logic from here: https://github.com/nrwl/nx/blob/master/packages/linter/src/utils/convert-tslint-to-eslint/project-converter.ts#L514

Reply:
Fixed the checking of the target, too, to be more proper! :)